macaroon.js

```diff
@@ -380,61 +369,62 @@ Macaroon.prototype.addCaveat = function(caveatId, verificationId, loc) {
 //
 // Verify throws an exception if the verification fails.
 Macaroon.prototype.verify = function(rootKey, check, discharges) {
-	rootKey = makeKey(rootKey)
+	rootKey = makeKey(rootKey);
 	var used = [];
 	var i;
-	for(i in discharges){
-		used[i] = 0;
+	discharges = discharges || [];
+	for (i = 0; i < discharges.length; i++) {
+    used[i] = 0;
 	}
 	this._verify(this._signature, rootKey, check, discharges, used);
-	for(i in discharges){
-		var dm = discharges[i];
-		if(used[i] === 0){
-			throw new Error("discharge macaroon " + quote(dm.id()) + " was not used");
+	discharges.forEach(function(dm, i) {
+		if (used[i] === 0) {
+			throw new Error('discharge macaroon ' + quote(dm.id()) + ' was not used');
 		}
-		if(used[i] !== 1){
+		if (used[i] !== 1) {
 			// Should be impossible because of check in verify1, but be defensive.
-			throw new Error("discharge macaroon " + quote(dm.id()) + " was used more than once");
+			throw new Error('discharge macaroon ' + quote(dm.id()) + ' was used more than once');
 		}
-	}
+	});
 };
 
 Macaroon.prototype._verify = function(rootSig, rootKey, check, discharges, used) {
 	var caveatSig = keyedHash(rootKey,  sjcl.codec.utf8String.toBits(this.id()));
 	var i, di;
-	for(i in this._caveats){
-		var cav = this._caveats[i];
-		if(cav._vid !== null){
+	this._caveats.forEach(function(cav) {
+		if (cav._vid !== null) {
 			var cavKey = decrypt(caveatSig, cav._vid);
 			var found = false;
-			for(di in discharges){
-				var dm = discharges[di];
-				if(dm.id() !== cav._identifier) {
+			var di, dm;
```

Comment:
Shouldn't these vars be at the top level of the function, or is it considered OK in this case with a single if?